### PR TITLE
Feature/native net liteserv rebased

### LINKED ISSUE #508: Support native .NET listener
I got a stripped down P2P test running on Windows and discovered many library incompatibilities, the biggest blocker is ansible. There are a few options.

1. Make testkit work on windows (We may have to use cygwin, http://www.jeffgeerling.com/blog/running-ansible-within-windows)
1. Use existing mac controller and provision a .NET windows node. http://docs.ansible.com/ansible/intro_windows.html. I think this is the best approach.

### PR DESCRIPTION
**Pull request by sethrosetter**
Fixes #508 

TODO:
- [x] Support all storage engines
- [x] Setup up automation
  - [x] Need CentOS7 agent to execute the tests / host sync_gateway (in same network to allow bidirectional replication)
http://uberjenkins.sc.couchbase.com/job/pytest_net-windows-listener-client-sync-gateway

- Support for provisioning window machine with LiteServ .NET
- Update Readme for instructions on how to prepare windows box
- Add support for running tests against .NET windows

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -657,6 +658,45 @@ $ sudo pfctl -ef /etc/pf.conf
...
+ansible_user=FakeUser
+ansible_password=FakePassword
+ansible_port=5986
+ansible_connection=winrm
+# The following is necessary for Python 2.7.9+ when using default WinRM self-signed certificates:
+ansible_winrm_server_cert_validation=ignore
+```
+Save as `resources/cluster_configs/windows`
+
+### IMPORTANT: Do not publish or check this in. If you do, anyone could potentially access your machine.
```
Comment (tleyden) [inline README.md]: This comment is a bit confusing.  As the reader, what exactly am I not supposed to be checking in?

File: README.md [hunk 2]
```diff
@@ -657,6 +658,45 @@ $ sudo pfctl -ef /etc/pf.conf
...
+
+- Download and execute this in the windows target PowerShell (Run as Administrator)
+https://github.com/ansible/ansible/blob/devel/examples/scripts/ConfigureRemotingForAnsible.ps1
```
Comment (tleyden) [inline README.md]: Can you name the links?  Eg, [ConfigureRemotingForAnsible Script](https://github.com/ansible/ansible/blob/devel/examples/scripts/ConfigureRemotingForAnsible.ps1)

File: keywords/LiteServ.py [hunk 3]
```diff
@@ -587,6 +725,15 @@ def verify_liteserv_launched(self, platform, host, port, version_build):
...
                 raise LiteServError("Expected version does not match actual version: Expected={}  Actual={}".format(version_build, running_version_composed))
+        elif is_net_win:
+            # running_version_parts format:
+            #  ['.NET', 'Microsoft', 'Windows', '10', 'Enterprise', 'x64', '1.4.0', 'build0043', '5cfe25b']
+            running_version_parts = re.split("[ /-]", lite_version)
+            version = running_version_parts[6]
+            build = int(running_version_parts[7].strip("build"))
+            running_version_composed = "{}-{}".format(version, build)
+            if version_build != running_version_composed:
+                raise LiteServError("Expected version does not match actual version: Expected={}  Actual={}".format(version_build, running_version_composed))
```
Comment (tleyden) [inline keywords/LiteServ.py]: Can you add a unit test for this?

File: keywords/MobileRestClient.py [hunk 4]
```diff
@@ -1073,11 +1073,27 @@ def start_replication(self,
...
+                "continuous": continuous,
+                "source": {"url": source},
+                "target": {"url": target}
+            }
+
+            if to_auth is not None:
+                data["target"]["headers"] = {"Cookie": to_auth}
+
+            if from_auth is not None:
+                data["source"]["headers"] = {"Cookie": from_auth}
```
Comment (tleyden) [inline keywords/MobileRestClient.py]: Could be less duplication with:

```
auth = None
if to_auth is not None:
   auth = to_auth
if from_auth is not None:
   auth = from_auth
data["target"]["headers"] = {"Cookie": auth}
```

File: setup.sh [hunk 5]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/bin/bash
```
Comment (tleyden) [inline setup.sh]: I think `/usr/bin/env bash` is more robust to `bash` being in different locations.  What's the reasoning behind this change?

File: keywords/LiteServ.py [hunk 6]
```diff
@@ -113,7 +115,7 @@ def get_download_url(self, platform, version_build, storage_engine):
                 url = "{}/couchbase-lite-android/release/{}/{}/{}".format(LATEST_BUILDS, version, version_build, file_name)
             else:
                 url = "{}/couchbase-lite-android/{}/{}/{}".format(LATEST_BUILDS, version, version_build, file_name)
-        elif platform == "net":
+        elif platform == "net" or platform == "net-win":
```
Comment (sethrosetter) [inline keywords/LiteServ.py]: I will address this as part of #722

File: README.md (hunk 1 above)
Comment (sethrosetter) [inline README.md]: Done

File: README.md (hunk 2 above)
Comment (sethrosetter) [inline README.md]: Done

File: keywords/MobileRestClient.py (hunk 4 above)
Comment (sethrosetter) [inline keywords/MobileRestClient.py]: data["source"] and data["target"] are potentially different values

File: setup.sh (hunk 5 above)
Comment (sethrosetter) [inline setup.sh]: I was hitting some Jenkins issues, but resolved them on the Jenkins side

Comment (sethrosetter) [inline setup.sh]: Reverted the change

**Pull request merged**